src/droid.cpp

```diff
@@ -2626,7 +2651,9 @@ void giveExperienceForSquish(DROID *psDroid)
 	{
 		const uint32_t expGain = std::max(65536 / 2, 65536 * getExpGain(psDroid->player) / 100);
 		droidIncreaseExperience(psDroid, expGain);
-		cmdDroidUpdateExperience(psDroid, expGain);
+		if (droidWithinCommanderRange(psDroid, false)) {
```

Comment:
This call should probably instead be inside `cmdDroidUpdateExperience`, within the `if hasCommander(psShooter)` block